src/core/reader/types/mod.rs

```diff
@@ -201,6 +201,86 @@ impl WasmReadable for FuncType {
     }
 }
 
+/// <https://webassembly.github.io/spec/core/binary/instructions.html#binary-blocktype>
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BlockType {
+    Empty,
+    Returns(ValType),
+    Type(u32),
+}
+
+impl WasmReadable for BlockType {
+    fn read(wasm: &mut WasmReader) -> Result<Self> {
+        // FIXME: Use transactions for ValType::read
+        if wasm.peek_u8()? as i8 == 0x40 {
+            // Empty block type
+            let _ = wasm.read_u8().expect("read to succeed, as we just peeked");
```

Comment:
`.unwrap_validated` imho, you can provide the reason as a comment, though it is pretty self-explanatory